primer/brand
# Designer / Developer Integration Grid

## Summary (WIP / Proposal)

This adds a visual layer to display grid-columns as an overlay, overtop our markup. This is similar to how Figma allows us to toggle the grid overlay inside of our designs. I feel this could greatly benefit developer by speeding up the development time and potentially shortening the feedback loop:
- Assist in creating faithful translation of source design into code
- Help eliminate guesswork (Does this line up correctly?)  by placing items precisely on the grid
- Eliminate or reduce the need for alignment feedback in the Q&A process
- Improve integration between stakeholder, designer and developer

#### Not yet added:
- Only show in dev environments
- Add tests

#### Additional Props Discussion: (What should we add?)
- maxWidth
- gutterSize (with breakpoints?)
- baseProps
  - className
  - id
  - ref

## What should reviewers focus on?

- What should be added?
- Anything added that's unnecessary?
- Any defaults should change?

## Steps to test:

1. View:
    - DevGrid Component Story
    - KitchenSink Example Layout
    - Resources Example Layout
2. Click inside the story (to give the iFrame focus) and type `control + shift + L`
3. Verify that
    - You can toggle the grid
    - It does not interfere with any mouse events

## Supporting resources

- For https://github.com/primer/brand/issues/99

## Screenshots

https://user-images.githubusercontent.com/9965014/199043921-a9b2bc18-9a2c-4874-ab40-3d7215b3b446.mov

<img width="600" alt="image" src="https://user-images.githubusercontent.com/9965014/199044280-a8b91686-a12f-4e40-8a99-d176ae68aa9a.png">
